File: leet/array_string/string_compression_443.java
class Solution {
    public int compress(char[] chars) {
        int write = 0;
        int anchor = 0;

        for (int read = 0; read < chars.length; read++) {
            if (read + 1 == chars.length || chars[read] != chars[read + 1]) {
                chars[write++] = chars[anchor];

                int runLen = read - anchor + 1;
                if (runLen > 1) {
                    for (char d : String.valueOf(runLen).toCharArray()) {
                        chars[write++] = d;
                    }
                }
                anchor = read + 1;
            }
        }
        return write;
    }
}
// 這不是自己寫的，看不懂

File: leet/array_string/string_compression_443.py
from typing import List

class Solution:
    def compress(self, chars: List[str]) -> int:
        result = []
        counter = 1
        for i in range(len(chars)):
            if i<len(chars)-1 and chars[i] == chars[i+1]:
                counter+=1
            else:
                result.append(chars[i])
                if counter > 1:
                    result.extend(list(str(counter)))
                counter = 1

        
        chars.clear()
        chars.extend(result)
        return len(result)
    
    # ---------------------------------------------------------------
    # 下面這作法出問題在於他會一次count()全部的內容物
    def compress(self, chars: List[str]) -> int:
        result = []
        for n in chars:
            if n not in result:
                temp = chars.count(n)
                result.append(n)
                if temp != 1:
                    print(list(str(temp)))
                    result.extend(list(str(temp)))
        
        chars.clear()
        chars.extend(result)
        return len(result)


aa = Solution()
print(aa.compress(["a","b","b","b","b","b","b","b","b","b","b","b","b"]))


# print(result.items()) 會得到
# dict_items([('a', 2), ('b', 2), ('c', 3)])